crates/forge-runner/src/printing.rs

```diff
@@ -1,59 +1,60 @@
-use crate::test_case_summary::TestCaseSummary;
+use crate::{test_case_summary::TestCaseSummary, test_crate_summary::AnyTestCaseSummary};
 use console::style;
 
-pub(crate) fn print_test_result(test_result: &TestCaseSummary) {
-    if let TestCaseSummary::Skipped { .. } = test_result {
-        return;
+fn result_header(any_test_result: &AnyTestCaseSummary) -> String {
+    if any_test_result.is_passed() {
+        return format!("[{}]", style("PASS").green());
     }
+    if any_test_result.is_failed() {
+        return format!("[{}]", style("FAIL").red());
+    }
+    if any_test_result.is_ignored() {
+        return format!("[{}]", style("IGNORE").yellow());
+    }
+    unreachable!()
+}
 
-    let result_header = match test_result {
-        TestCaseSummary::Passed { .. } => format!("[{}]", style("PASS").green()),
-        TestCaseSummary::Failed { .. } => format!("[{}]", style("FAIL").red()),
-        TestCaseSummary::Ignored { .. } => format!("[{}]", style("IGNORE").yellow()),
-        TestCaseSummary::Skipped { .. } => {
-            unreachable!()
-        }
-    };
-
-    let result_name = match test_result {
-        TestCaseSummary::Ignored { name }
-        | TestCaseSummary::Failed { name, .. }
-        | TestCaseSummary::Passed { name, .. } => name,
-        TestCaseSummary::Skipped {} => {
-            unreachable!()
-        }
-    };
-
-    let result_message = match test_result {
-        TestCaseSummary::Passed { msg: Some(msg), .. } => format!("\n\nSuccess data:{msg}"),
-        TestCaseSummary::Failed { msg: Some(msg), .. } => format!("\n\nFailure data:{msg}"),
-        _ => String::new(),
-    };
-
-    let fuzzer_report = match test_result.runs() {
-        None => String::new(),
-        Some(runs) => {
-            if matches!(test_result, TestCaseSummary::Failed { .. }) {
-                let arguments = test_result.arguments();
-                format!(" (fuzzer runs = {runs}, arguments = {arguments:?})")
-            } else {
-                format!(" (fuzzer runs = {runs})")
+pub(crate) fn print_test_result(any_test_result: &AnyTestCaseSummary) {
```

Comment:
It'd be neat if those 2 could be part of a trait

Reply:
not sure wdym by  `those 2` and what trait do you mean

Reply:
The two functions that are ran with `AnyTestCaseSummary` as an argument, they could be a trait of their own (like `AnyTestCaseSummaryDisplay` perhaps?)

Reply:
Imo outside of the scope of this PR, also don't see a point in implementing trait if there is no generalization among different types.

Reply:
that's not crucial though

Reply:
I am not sure if I like the idea, if it is not crucial I will skip this one (I generally dont like calling print on an object)